bug: int arithmetic capped at int64 despite arbitrary precision storage

## Summary

The `int` type can store arbitrary precision values (backed by `big.Int`), but arithmetic operations incorrectly check against int64 bounds and fail with overflow errors.

## Steps to Reproduce

**Storage works (arbitrary precision):**
```ez
import @std
using std

do main() {
    temp big int = 999999999999999999999999999999999999999999999999999999999999
    println("Big: ${big}")  // Works!
}
```

**Arithmetic fails (int64 cap):**
```ez
import @std
using std

do main() {
    temp a int = 999999999999999999999999999999
    temp b int = a + a  // ERROR: exceeds int64 range
}
```

```
error[E5005]: integer overflow: 999999999999999999999999999999 + 999999999999999999999999999999 exceeds int64 range
```

Even simple operations fail:
```ez
temp x int = 999999999999999999999999999999 + 1  // ERROR
```

## Expected Behavior

Since `int` uses `big.Int` internally, all arithmetic operations should also use arbitrary precision:

```ez
temp a int = 999999999999999999999999999999
temp b int = a + a  // Should work: 1999999999999999999999999999998
temp c int = a * a  // Should work: massive number
```

## Affected Operations

- Addition (`+`)
- Multiplication (`*`)
- Likely subtraction and others too

## Root Cause

The evaluator is likely converting `big.Int` to `int64` before performing arithmetic operations, then checking for overflow. The fix should use `big.Int` methods directly (`Add`, `Mul`, etc.) without int64 conversion.

--- a/pkg/interpreter/evaluator.go
+++ b/pkg/interpreter/evaluator.go
@@ -62,15 +62,19 @@ func init() {
 }
 
 // getTypeBounds returns the min and max values for a given integer type
+// Returns nil, nil for arbitrary precision types (int, uint) that have no bounds
 func getTypeBounds(typeName string) (min, max *big.Int) {
 	switch typeName {
+	case "int", "uint":
+		// Arbitrary precision - no bounds
+		return nil, nil
 	case "i8":
 		return minInt8, maxInt8
 	case "i16":
 		return minInt16, maxInt16
 	case "i32":
 		return minInt32, maxInt32
-	case "i64", "int", "":
+	case "i64", "":
 		return minInt64, maxInt64
 	case "i128":
 		return minInt128, maxInt128
@@ -82,7 +86,7 @@ func getTypeBounds(typeName string) (min, max *big.Int) {
 		return zero, maxUint16
 	case "u32":
 		return zero, maxUint32
-	case "u64", "uint":
+	case "u64":
 		return zero, maxUint64
 	case "u128":
 		return zero, maxUint128
@@ -95,8 +99,13 @@ func getTypeBounds(typeName string) (min, max *big.Int) {
 }
 
 // checkOverflow checks if a value is within bounds for a given type
+// Returns false for arbitrary precision types (int, uint) that have no bounds
 func checkOverflow(result *big.Int, typeName string) bool {
 	min, max := getTypeBounds(typeName)
+	if min == nil || max == nil {
+		// Arbitrary precision type - no overflow possible
+		return false
+	}
 	return result.Cmp(min) < 0 || result.Cmp(max) > 0
 }
 

--- a/pkg/interpreter/evaluator_test.go
+++ b/pkg/interpreter/evaluator_test.go
@@ -4132,7 +4132,7 @@ func TestPostfixOverflowDetection(t *testing.T) {
 		{
 			name: "increment overflow",
 			input: `
-temp i int = 9223372036854775807
+temp i i64 = 9223372036854775807
 i++
 `,
 			expectError: true,
@@ -4141,7 +4141,7 @@ i++
 		{
 			name: "decrement underflow",
 			input: `
-temp i int = -9223372036854775808
+temp i i64 = -9223372036854775808
 i--
 `,
 			expectError: true,
@@ -4276,8 +4276,8 @@ func TestLargeIntegerOverflow(t *testing.T) {
 			"E5006",
 		},
 		{
-			"int overflow still works",
-			"temp max int = 9223372036854775807\ntemp r int = max + 1",
+			"i64 overflow still works",
+			"temp max i64 = 9223372036854775807\ntemp r i64 = max + 1",
 			"E5005",
 		},
 	}

